RoddyCore/src/de/dkfz/roddy/plugins/LibrariesFactory.groovy

```diff
@@ -238,16 +270,21 @@ public class LibrariesFactory extends Initializable {
             logger.postSometimesInfo("Parsing plugins folder: ${pBaseDirectory}");
             if (!pBaseDirectory.exists()) {
                 logger.severe("The plugins directory $pBaseDirectory does not exist.")
+                mapOfErrorsForPluginFolders.get(pBaseDirectory, []) << "The plugins directory $pBaseDirectory does not exist.".toString()
                 continue;
             }
             if (!pBaseDirectory.canRead()) {
                 logger.severe("The plugins directory $pBaseDirectory is not readable.")
+                mapOfErrorsForPluginFolders.get(pBaseDirectory, []) << "The plugins directory $pBaseDirectory is not readable.".toString()
             }
 
             File[] directoryList = pBaseDirectory.listFiles().sort() as File[];
             for (File pEntry in directoryList) {
 
-                def workflowType = determinePluginType(pEntry)
+                Map<Integer, List<String>> errors = [:]
+                def workflowType = determinePluginType(pEntry, errors)
+                mapOfErrorsForPluginEntries[pEntry.path] = (errors[1] + errors[2])
```

Comment:
That's not intuitive. Why only the 2nd and 3rd error? The code does document this somehow.